stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpConnectionManager.java

```diff
@@ -66,12 +69,8 @@ private HttpConnectionManager() {
         serverBootstrapConfiguration = HTTPConnectorUtil
                 .getServerBootstrapConfiguration(trpConfig.getTransportProperties());
 
-        if (isHTTPTraceLoggerEnabled()) {
-            try {
-                ((BLogManager) BLogManager.getLogManager()).setHttpTraceLogHandler();
-            } catch (IOException e) {
-                throw new BallerinaConnectorException("Error in configuring HTTP trace log");
-            }
+        if (HttpConnectionManager.getInstance().isHTTPTraceLoggerEnabled()) {
```

Comment:
Leave this as `isHTTPTraceLoggerEnabled()`. Otherwise you still run into the same issue as before.